okhttp/src/main/java/okhttp3/internal/http1/Http1Codec.java

```diff
@@ -491,4 +516,18 @@ private void readChunkSize() throws IOException {
       closed = true;
     }
   }
+
+  private class Http1ConnectionDeadline extends AsyncTimeout {
+    @Override protected void timedOut() {
+      state = STATE_CANCEL;
```

Comment:
I don’t think this is safe concurrency wise. Instead we want to manually break the connection by calling cancel().